python/ray/tune/utils/util.py

```diff
@@ -484,6 +486,14 @@ def atomic_save(state: Dict, checkpoint_dir: str, file_name: str,
     os.replace(tmp_search_ckpt_path, os.path.join(checkpoint_dir, file_name))
 
 
+def find_free_port():
```

Comment:
Is there a convention for where new functions should be, e.g. end of file?

Reply:
No convention; usually locally grouped by similarity